internal/pkg/service/common/httpserver/middleware/otel_apdex.go

```diff
@@ -0,0 +1,70 @@
+package middleware
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"go.opentelemetry.io/otel/metric"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/telemetry"
+)
+
+// apdexCounter is a helper to calculate apdex metric and process it by OpenTelemetry.
+
+// Apdex[T] is a value between 0.0-1.0, it is defined by the formula:
+// = (satisfied_requests_count + tolerated_requests_count/2) / total_requests_count
+//
+// Ranges:
+// - Satisfied request:  no server error AND duration < T
+// - Tolerated request:  no server error AND T < duration < 4T
+// - Frustrated request: server error OR duration > 4T
+//
+// As a metric, it is calculated cumulatively, therefore:
+// - Satisfied request  contributes with 1.0 to the total AVG value.
+// - Tolerated request  contributes with 0.5 to the total AVG value.
+// - Frustrated request contributes with 0.0 to the total AVG value.
+type apdexCounter struct {
+	counter    metric.Int64Counter
+	thresholds []*apdexThreshold
+}
+
+type apdexThreshold struct {
+	sum         metric.Float64Counter
+	satisfiedMs float64
+	toleratedMs float64
+}
+
+func newApdexCounter(meter metric.Meter, thresholds []time.Duration) *apdexCounter {
+	out := &apdexCounter{counter: telemetry.MustInstrument(meter.Int64Counter("keboola_go_http_server_apdex_count"))}
```

Comment:
Akoát ten counter je per worker, ne? DD pak řeší apdex per worker nebo dělá nějaký sum/avg?

Reply:
Apdex je pre API, .... resp. `httpserver`. Kazda API node generuje svoje Prometheus metriky na `:9000/metrics`. Datadog ich scrapuje a spoji dokopy - spocita. 

Preto sa samostatne uvadza `count` a `sum`, aby si ich DataDog mohol spocitat a vysledkom je vazeny priemer AVG.

Aj ked ma metriku typu histogram, tak vo vysledku je to viac metrik a medzi nimi aj `..._sum` a `..._count`.